PYTHON_CDM_Conversion_code/monthly_to_cdm_all_v1.py

```diff
@@ -1,635 +1,708 @@
+            df['date_time'] = df['date_time'].astype('str')
+            df.date_time = df.date_time + '+00'
+            
+            # extract precip
+            try:
+                dfprc = df[["observation_id","report_type","date_time","date_time_meaning",
+                          "latitude","longitude","observation_height_above_station_surface"
+                          ,"observed_variable","units","observation_value",
+                          "value_significance","observation_duration","platform_type",
+                          "station_type","primary_station_id","station_name","quality_flag"
+                          ,"data_policy_licence"]]
+                
+                # change for each variable to convertto cdm compliant values
+                dfprc["observation_value"] = df["PRCP"]
+                # change for each variable if required
+                dfprc["observation_height_above_station_surface"]="1"
+                dfprc["units"] = "710"
+                dfprc["observed_variable"] = "44"
+                # set up matching values for merging with record_id_month to add information source_id for first configuration only due yto lack of information
+                dfprc["record_number"] = "1"   
+                dfprc['primary_station_id_2'] = dfprc['primary_station_id'].astype(str)+'-'+dfprc['record_number'].astype(str)
+                dfprc["report_id"] = dfprc["date_time"]
+                
+                # merge with record_id_mnth.csv to add source id
+                df2 = pd.read_csv(utils.MONTHLY_STATION_RECORD_ENTRIES_MONTHLY_OBS_LITE, encoding='latin-1')
+                dfprc = dfprc.astype(str)
+                df2 = df2.astype(str)
+                dfprc = df2.merge(dfprc, on = ['primary_station_id_2'])
+                dfprc['observation_id'] = dfprc['primary_station_id'].astype(str)+'-'+dfprc['record_number'].astype(str)+'-'+dfprc['date_time'].astype(str)
+                dfprc['observation_id'] = dfprc['observation_id'].str.replace(r' ', '-')
+                # remove unwanted last twpo characters
+                dfprc['observation_id'] = dfprc['observation_id'].str[:-12]
+                dfprc["observation_id"] = dfprc["observation_id"]+'-'+dfprc['observed_variable'].astype(str)+'-'+dfprc['value_significance'].astype(str)
+                
+                # reorder columns and drop unwanted columns
+                dfprc = dfprc[["observation_id","report_type","date_time","date_time_meaning",
+                          "latitude","longitude","observation_height_above_station_surface"
+                          ,"observed_variable","units","observation_value",
+                          "value_significance","observation_duration","platform_type",
+                          "station_type","primary_station_id","station_name","quality_flag"
+                          ,"data_policy_licence","source_id","primary_station_id_2"]]
+            except:
```

Comment:
Should in #5 address these "naked" except statements

Reply:
Not sure what you mean here?

Reply:
It was a note that rather than a generalised `except` statement (which catches everything, but with the risk that it covers up for things we want to know about) we should in Issue #5 replace these with `except XYZError` statements which are specific to the failures we're expecting.